python/ray/dag/dag_node_operation.py

```diff
@@ -72,38 +76,55 @@ def __init__(
         # be READ, COMPUTE, or WRITE.
         self.in_edges: Set[Tuple[int, _DAGNodeOperationType]] = set()
         self.out_edges: Set[Tuple[int, _DAGNodeOperationType]] = set()
+        # The collective group is a set of nodes that belong to the same
+        # collective operation. Each node is represented by a tuple of its
+        # task idx and type.
+        self.collective_group: Set[Tuple[int, _DAGNodeOperationType]] = set()
+        # The ready collective nodes are the nodes that are ready to be executed,
+        # i.e., their in-degrees are zero. When a collective node is ready, it
+        # will be added to the ready collective nodes of all the nodes in its
+        # collective group.
+        self.ready_collective_nodes: Set[Tuple[int, _DAGNodeOperationType]] = set()
 
-    @property
-    def in_degree(self) -> int:
-        return len(self.in_edges)
+    def __repr__(self):
+        return (
+            f"_DAGOperationGraphNode("
+            f"operation: {self.operation}, "
+            f"task_idx: {self.task_idx}, "
+            f"actor_handle: {self.actor_handle}, "
+            f"requires_nccl: {self.requires_nccl})"
+        )
 
     def __lt__(self, other: "_DAGOperationGraphNode"):
         """
         This function defines the order of the nodes in the priority queue used in
         `_select_next_nodes`. The priority queue is a min-heap, so the node with
         higher priority is considered "less than" the other node.
         """
-        # If two nodes belong to the same actor, select the one with
-        # the smaller `exec_task_idx`.
+
+        # Compare two nodes based on the following rules:
+        # 1. If lhs and rhs have different `exec_task_idx`, the one with the smaller
+        #    `exec_task_idx` has higher priority.
+        # 2. If lhs and rhs have the same `exec_task_idx`, the one with the smaller
+        #    `task_idx` has higher priority.
```

Comment:
Can you improve this comment to explain the intuition behind the priority? (I realize this part is mostly copied, but the comment text doesn't add much beyond the code itself)

Reply:
Done.